src/main/java/jarvis/logic/commands/AddConsultCommand.java

```diff
@@ -72,7 +72,14 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Student> lastShownList = model.getFilteredStudentList();
 
-        Set<Student> studentSet = new HashSet<>();
+        Set<Student> studentSet = new TreeSet<>((s1, s2) -> {
+            int result = s1.getName().toString().toLowerCase().compareTo(s2.getName().toString().toLowerCase());
+            if (result == 0) {
+                return s1.getMatricNum().toString().compareTo(s2.getMatricNum().toString());
+            }
+            return result;
+        });
+
```

Comment:
I think the code to compare between students is abit repetitive across the 5 changed files. Maybe we can shift this implementation to the student class? 

So I am thinking the `Student` class can `implement Comparable<Student>` then shift this comparison to the `compareTo` method. Then there would be no need to specify an explicit Comparator and the comparison between students can be placed in the student class.